### PR TITLE
Add Elasticsearch Topology Cluster Docs

### PR DESCRIPTION
**Pull request by DRoyyy**
Signed-off-by: DRoyyy <diptaroy@appscode.com>

### CODE REVIEW

File: docs/guides/elasticsearch/clustering/topology-cluster/simple-dedicated-cluster/yamls/es-cluster.yaml [hunk 1]
```diff
@@ -1,15 +1,21 @@
 apiVersion: kubedb.com/v1alpha2
 kind: Elasticsearch
 metadata:
-  name: es-topology
+  name: es-cluster
   namespace: demo
 spec:
   enableSSL: true 
-  version: searchguard-7.9.3
```
Comment (kamolhasan) [inline docs/guides/elasticsearch/clustering/topology-cluster/simple-dedicated-cluster/yamls/es-cluster.yaml:8]: Use `searchguard-7.14.2`

File: docs/guides/elasticsearch/clustering/topology-cluster/simple-dedicated-cluster/yamls/es-cluster.yaml [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
+  version: xpack-7.14.0
   storageType: Durable
   topology:
     master:
       replicas: 2
+      resources:
```
Comment (kamolhasan) [inline docs/guides/elasticsearch/clustering/topology-cluster/simple-dedicated-cluster/yamls/es-cluster.yaml:13]: Remove resources

**Review by kamolhasan — CHANGES_REQUESTED** [review]

File: docs/guides/elasticsearch/clustering/topology-cluster/hot-warm-cold-cluster/index.md [hunk 3]
```diff
@@ -0,0 +1,580 @@
...
+    weight: 15
+menu_name: docs_{{ .version }}
+section_menu_id: guides
+---
+
+> New to KubeDB? Please start [here](/docs/README.md).
+
+# Elasticsearch Hot-Warm-Cold Cluster
+
+Hot-warm-cold architectures are common for time series data such as logging or metrics. For example, assume Elasticsearch is being used to aggregate log files from multiple systems. Logs from today are actively being indexed and this week's logs are the most heavily searched (hot). Last week's logs may be searched but not as much as the current week's logs (warm). Last month's logs may or may not be searched often, but are good to keep around just in case (cold).
```
[the remark this answers is not in the document]
Comment (DRoyyy) [reply docs/guides/elasticsearch/clustering/topology-cluster/hot-warm-cold-cluster/index.md]: solved

File: docs/guides/elasticsearch/clustering/topology-cluster/hot-warm-cold-cluster/index.md [hunk 4]
```diff
@@ -0,0 +1,580 @@
...
+standard (default)                      rancher.io/local-path     Delete          WaitForFirstConsumer   false                  10m
+linode-block-storage                    linodebs.csi.linode.com   Delete          Immediate              true                   10m
+linode-block-storage-retain (default)   linodebs.csi.linode.com   Retain          Immediate              true                   10m
+```
+
+Here, we use `linode-block-storage` as StorageClass in this demo.
+
+## Create Elasticsearch Hot-Warm-Cold Cluster
+
+We are going to create a Elasticsearch Hot-Warm-Cold cluster in topology mode. Our cluster will be consist of 2 master nodes, 2 ingest nodes, 1 data content node, 3 data hot nodes, 2 data warm node, and 2 data cold nodes. Here, we are using Elasticsearch version ( `xpack-7.14.0` ) of ElasticStack distribution for this demo. To learn more about the Elasticsearch CR, visit [here](/docs/guides/elasticsearch/concepts/elasticsearch/index.md).
```
[the remark this answers is not in the document]
Comment (DRoyyy) [reply docs/guides/elasticsearch/clustering/topology-cluster/hot-warm-cold-cluster/index.md]: solved

**Review by kamolhasan — APPROVED** [review]
LGTM!

**Pull request merged**